### PR TITLE
Consensus Message Submission Prehandle

### LINKED ISSUE #4810: ConsensusSubmitMessageHandler.preHandle()

### PR DESCRIPTION
**Pull request by mhess-swl**
Closes #4810

### CODE REVIEW

**Review by iwsimon — DISMISSED** [review]
LGTM

**Review by povolev15 — DISMISSED** [review]
LGTM

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java [hunk 1]
```diff
@@ -43,20 +43,30 @@ public ConsensusSubmitMessageHandler() {}
...
+
+        final var op = context.getTxn().getConsensusSubmitMessage();
+        final var topicMeta = topicStore.getTopicMetadata(op.getTopicID());
+        if (topicMeta.failed()) {
+            context.status(ResponseCodeEnum.INVALID_TOPIC_ID);
+            return;
+        }
+
+        final var submitKey = topicMeta.metadata().submitKey();
+        if (submitKey.isEmpty()) {
```
Comment (tinker-michaelj) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java:65]: This isn't current behavior, is it?

**Review by iwsimon — DISMISSED** [review]
LGTM

**Review by tinker-michaelj — CHANGES_REQUESTED** [review]
Seems like this changes `ConsensusSubmitMessage` semantics to require the topic has a `submitKey`?

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java (hunk 1 above)
Comment (mhess-swl) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java:65]: Yep you're right, not sure where I got this from. Good catch, thanks!

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Review by iwsimon — APPROVED** [review]
LGTM

**Pull request merged**